### ISSUE TITLE
"Win Chance" display during deck optimization is unclear

### ISSUE THREAD

**Issue by nivomi** [MgAl2O4/FFTriadBuddy#60]
This is kind of a two-for one thing - when using the deck optimization, I sometime see "win chance: "xx% (DRAW)", and I'm not exactly sure what that means relative to just "win chance: xx%".

Also, while calculating, I sometimes see the tool pick a deck and display a winrate, and then switch to a different deck with a worse displayed winrate. I can't tell whether this is because it's optimizing for something I can't see, or if it's a bug that's making it choose worse decks?

Ex: https://user-images.githubusercontent.com/651417/127047863-8aadffa4-310d-4e7e-888a-07173d9d5a29.mp4

STR:
1. Use the following config (zipped because github won't let me just upload the json)
[FFTriadBuddy-settings.zip](https://github.com/MgAl2O4/FFTriadBuddy/files/6880737/FFTriadBuddy-settings.zip)
2. Try to optimize for Kiuka with no regional rules set.
3. Observe that the display of winrates during optimization is a bit confusing/unclear.

Let me know if there's anything else I can provide or otherwise assist - this project is extremely impressive and useful, and I hope this report is helpful!

**Comment by MgAl2O4** [comment]
Thanks for detailed repro! 

Switching to lower chance does look like a bug and it's not supposed to lower the score. Only exception would be draw going down by max 10% to win, which is not the case on your movie. Will investigate.

I'm kind of on the edge for splitting result tree way: win - draw - lose. Showing draw chance basically means that you have higher chance of draw than winning, which allows me to get away with showing higher number to player :P That being said, ultimately the goal is to win, so it's just unnecessary noise. Seems like a good candidate to expose in options for people to tweak if they want to.

**Comment by MgAl2O4** [comment]
TL;DR - deck optimizer behavior is actually intentional and differences are negligible during match

Long(er) explanation:
Solver is using very crude version of MCTS algorithm to estimate chances, evaluating just at first branch. It's sacrificing accuracy for speed, but as game progresses the accuracy goes up (less choices to make). There's enough randomness in play (npc behavior and/or more chaotic modifiers) that is works out good enough.

Deck optimizer is similar, but not the same. To squeeze in more simulations, it's not even fully evaluating all actions on first branch, but instead just runs a few thousand of games and checks how many ended up as win and draw. Sample is just a tiny fraction of play space but it's an acceptable compromise between optimization time and accuracy. Next, cards selected by optimizer are passed along as "player deck" and solver mentioned above does it's own calculations to produce win chance. Different solver = different result, neither of them representing entire play space. Just various approximations.

Only way to make it better would be unifying solvers (and keeping all randomness in sync), or replacing Monte Carlo approach is some ML. I'm bad at ML :<

**Comment by nivomi** [comment]
That all makes sense, and that's basically the answer I expected - ML is hard, and I can't really think of a great way to 'shape the problem' so that ML would effectively solve decks better than the basic shuffler.

Maybe just adding a tilde to the winrate when it's still calculating would help suggest that it isn't a 'final' calculation?

**Comment by MgAl2O4** [comment]
Fixed in v89
(optimizer results are double checked with solver, will occasionally go back and forth with decks)
